apps/workspace-host/src/interface.js

```diff
@@ -496,35 +496,34 @@ function _checkServer(workspace, callback) {
 
   const startTime = new Date().getTime();
   function checkWorkspace() {
-    request(
-      {
-        url: `http://${workspace_server_settings.server_to_container_hostname}:${workspace.launch_port}/`,
-        timeout: healthCheckTimeout,
-      },
-      function (err, res, _body) {
-        if (err) {
-          // Do nothing, because errors are expected while the container is launching.
-        }
-        if (res && res.statusCode) {
-          // We might get all sorts of strange status codes from the server.
-          // This is okay since it still means the server is running and we're
-          // getting responses.
-          callback(null, workspace);
+    const abortController = new AbortController();
+    const abortTimeout = setTimeout(() => abortController.abort(), healthCheckTimeout);
```

Comment:
Based on https://www.npmjs.com/package/node-fetch#request-cancellation-with-abortsignal

Reply:
I typically reach for [`AbortSignal.timeout()`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout_static) to do this.

Reply:
I was not aware of it. Replaced in bd7460f9a.